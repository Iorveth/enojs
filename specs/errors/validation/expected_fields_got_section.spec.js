const eno = require('../../../eno.js');

const input = `
# language
eno: eno notation

# language
json: javascript object notation
`.trim();

describe('validation.expectedFieldsGotSection', () => {
  const document = eno.parse(input);

  let error;
  try {
    document.fields('language');
  } catch(err) {
    error = err;
  }

  it(`provides a correct message`, () => {
    expect(error.message).toMatchSnapshot();
  });

  it(`provides correct selection metadata`, () => {
    expect(error.selection).toMatchSnapshot();
  });
});
