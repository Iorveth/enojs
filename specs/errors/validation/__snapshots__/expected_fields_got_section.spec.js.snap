// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation.expectedFieldsGotSection provides a correct message 1`] = `
"Only fields with the name 'language' were expected, but a section with this name was found.

   Line | Content
 >    1 | # language
 >    2 | eno: eno notation
 >    3 | 
      4 | # language
      5 | json: javascript object notation
"
`;

exports[`validation.expectedFieldsGotSection provides correct selection metadata 1`] = `
Array [
  Array [
    0,
    0,
  ],
  Array [
    2,
    0,
  ],
]
`;
